File: kotlin/ktor-todo-list-api-openapi-first/src/main/kotlin/todolist/model/TodoModel.kt
package todolist.model

data class Todo(val id: Int, var title: String, var description: String, var state: TodoState)

data class TodoCreate(val title: String, val description: String, val state: TodoState)

data class TodoUpdate(val title: String?, val description: String?, val state: TodoState?)

enum class TodoState {
    OPEN,
    DONE,
}

File: typescript/express-zod-todo-list-api/src/todos/todo.model.ts
export type TodoState = "open" | "done";

export type TodoCreate = {
  title: string;
  description: string;
  state: TodoState;
};

export type TodoUpdate = {
  title?: string;
  description?: string;
  state?: TodoState;
};

export type Todo = {
  id: number;
  title: string;
  description: string;
  state: TodoState;
};
